dallinger/frontend/static/scripts/dallinger2.js

```diff
@@ -429,20 +449,30 @@ var dallinger = (function () {
 
   /**
    * Load an existing `Participant` into the dlgr.identity by making a ``POST``
-   * request to the experiment `/participant` route with an ``assignment_id``.
+   * request to the experiment `/participant` route with some ``assignment_info``
+   * which can be a scalar ``assignment_id`` or an object with ``entry_information``
+   * parameters
    *
    * @returns {jQuery.Deferred} See :ref:`deferreds-label`
    */
-  dlgr.loadParticipant = function(assignment_id) {
-    var deferred = $.Deferred(),
-        url = '/participant';
+  dlgr.loadParticipant = function(assignment_info) {
+    var data,
+        deferred = $.Deferred(),
+        url = '/load-participant';
+
+    if (typeof assignment_info == "object") {
```

Comment:
This `typeof` check will return True for `null`. Is that a problem here? Also, unless you're relying on implicit type coercion from `==` (hopefully not!), I'd model good behavior and use `===`.

Reply:
I should have used `===`, I think the handling of `null` values is probably fine. In one case we would post to the `/participant` route with `null` data resulting in a validation error, and the other we post with data `{assignment_id: null}` which has the same result.